### PR TITLE
chore: add identity feature for mssql

### PR DESCRIPTION
**Pull request by vmihailenco**

### CODE REVIEW

File: dialect/mysqldialect/dialect.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ func New() *Dialect {
 	d := new(Dialect)
 	d.tables = schema.NewTables(d)
 	d.features = feature.AutoIncrement |
+		feature.Identity | // for mssql dialect
```
Comment (vmihailenco) [inline dialect/mysqldialect/dialect.go:32]: @gandaldf this should be in mssql dialect

**Comment by gandaldf** [comment]
@vmihailenco This is more or less the same code I've wrote, my question is about field tags: we should tag them as "auto_increment" even if they are "identity"?
I'm new and I would like to understand the general philosophy of the project.
Soon I will publish my code, it's just a draft for now... nothing reliable.

**Comment by vmihailenco** [comment]
@gandaldf it is all about convenience. If you find a cleaner way to implement this - go ahead with it. But it might be a good idea to discuss the new way before going full in :)

Ultimately, we need some boolean to switch between "AUTO_INCREMENT" and "IDENTITY" in different dialects. `feature.Has` looks like a sensible way to do it.

>Soon I will publish my code, it's just a draft for now... nothing reliable.

Don't hesitate to ping me if you are stuck :+1:

**Comment by vmihailenco** [comment]
>I'm new and I would like to understand the general philosophy of the project.

Regarding philosophy... the goal is to find a common ground and features between Postgres/MySQL/MSSQL and provide uniform way to define models. So it must be the same option name to enable auto increments in different databases if they support the option, not different "branded" option names in different dialects.

So using `bun:",pk,autoincrement"` in all dialects is the right way forward if `identity` is just a different name for autoincrement.

**Pull request closed without merging**